docs/design/storage-auto-scale.md

```diff
@@ -0,0 +1,113 @@
+# Storage Automatic Scaler
+
+## Summary
+
+This feature would help users to get rid of maintenance task of keeping the system with enough available free space.
+In virtualized environments, we can scale the storage devices Osds so this feature helps in automatic scaling. 
+
+## Proposal
+
+Implement auto storage scale CR per device class of a specific Storagecluster.
+
+## High-Level Design
+
+- Create a StorageAutoScale CR per Storagecluster device class.
+- UI will create the StorageAutoScale CR with a unique name i.e. <storageclusterName>-<deviceClassName>.
+  
+    ```
+    apiVersion: ocs.openshift.io/v1
+    kind: StorageAutoScaler
+    metadata:
+        name: <storageclusterName>-<deviceClassName>
+        namespace: openshift-storage
+    spec:
+        storageCluster: 
+            name: <storageclusterName>
+        deviceClassName: ssd
+        deviceClassStorageCapacityLimit: <capacity>
```

Comment:
All settings are implicit to the device class, so we don't need the `deviceClass` as a prefix to the settings
```suggestion
        storageCapacityLimit: <capacity>
```

Reply:
after discussion we came to this, I started with `storageCapacityLimit` itself, I think no harm in putting prefix

Reply:
By the same argument, seems like then we need to name them `deviceClassMaxOsdSizeLimit`, or `deviceClassTimeoutSeconds`, and so on. So I would still vote to remove the prefix for consistency.